xwiki-platform-core/xwiki-platform-image/xwiki-platform-image-processing/xwiki-platform-image-processing-plugin/src/main/java/com/xpn/xwiki/plugin/image/ImagePlugin.java

```diff
@@ -367,33 +367,53 @@ private XWikiAttachment shrinkImage(XWikiAttachment attachment, int requestedWid
     private int[] reduceImageDimensions(int currentWidth, int currentHeight, int requestedWidth, int requestedHeight,
         boolean keepAspectRatio)
     {
-        double aspectRatio = (double) currentWidth / (double) currentHeight;
+        int[] dimensions;
+        // Keep the aspect ratio when requested or width or height are missing.
+        if (keepAspectRatio || requestedWidth <= 0 || requestedHeight <= 0) {
+            dimensions = reduceImageDimensions(currentWidth, currentHeight, requestedWidth, requestedHeight);
+        } else if (requestedWidth > currentWidth || requestedHeight > currentHeight) {
+            dimensions = reduceImageDimensions(requestedWidth, requestedHeight, currentWidth, currentHeight);
```

Comment:
I think this is quite nice but a comment to explain why this is done and why this works would be nice.

Reply:
I've improved to doc of `reduceImageDimensions`, let me know if that does not make things better.